shared/actions/chat2/index.js

```diff
@@ -1271,64 +1271,103 @@ const messageRetry = (action: Chat2Gen.MessageRetryPayload, state: TypedState) =
-    ? Constants.getConversationExplodingMode(state, conversationIDKey)
-    : 0
-  const ephemeralData = ephemeralLifetime !== 0 ? {ephemeralLifetime} : {}
-
-  // Inject pending message and make the call
-  const newMsg = Constants.makePendingTextMessage(
-    state,
-    conversationIDKey,
-    text,
-    Types.stringToOutboxID(outboxID.toString('hex') || ''), // never null but makes flow happy
-    ephemeralLifetime
-  )
-  const addMessage = Saga.put(
-    Chat2Gen.createMessagesAdd({
-      context: {type: 'sent'},
-      messages: [newMsg],
-    })
-  )
+let stellarConfirmPaymentModalShown = false
 
-  const postText = Saga.callUntyped(
-    RPCChatTypes.localPostTextNonblockRpcPromise,
-    {
-      ...ephemeralData,
-      body: text.stringValue(),
-      clientPrev,
-      conversationID: Types.keyToConversationID(conversationIDKey),
-      identifyBehavior: getIdentifyBehavior(state, conversationIDKey),
-      outboxID,
-      tlfName,
-      tlfPublic: false,
-    },
-    Constants.waitingKeyPost
-  )
+const messageSend = (action: Chat2Gen.MessageSendPayload, state: TypedState) =>
+  Saga.callUntyped(function*() {
+    const {conversationIDKey, text} = action.payload
+    const outboxID = Constants.generateOutboxID()
+    const meta = Constants.getMeta(state, conversationIDKey)
+    const tlfName = meta.tlfname
+    const clientPrev = Constants.getClientPrev(state, conversationIDKey)
 
-  // Do some logging to track down the root cause of a bug causing
-  // messages to not send. Do this after creating the objects above to
-  // narrow down the places where the action can possibly stop.
+    // disable sending exploding messages if flag is false
+    const ephemeralLifetime = flags.explodingMessagesEnabled
+      ? Constants.getConversationExplodingMode(state, conversationIDKey)
+      : 0
+    const ephemeralData = ephemeralLifetime !== 0 ? {ephemeralLifetime} : {}
+    const newMsg = Constants.makePendingTextMessage(
+      state,
+      conversationIDKey,
+      text,
+      Types.stringToOutboxID(outboxID.toString('hex') || ''), // never null but makes flow happy
+      ephemeralLifetime
+    )
 
-  logger.info('[MessageSend]', 'non-empty text?', text.stringValue().length > 0)
+    const addMessage = (p, response) => [
+      Saga.put(
+        Chat2Gen.createMessagesAdd({
+          context: {type: 'sent'},
+          messages: [newMsg],
+        })
+      ),
+      Saga.callUntyped(function() {
+        response && response.result()
+      }),
+    ]
+    const onShowConfirm = () => {
+      stellarConfirmPaymentModalShown = true
+      return [
+        Saga.put(Chat2Gen.createClearPaymentConfirmInfo()),
+        Saga.put(
+          RouteTreeGen.createNavigateAppend({
+            path: ['paymentsConfirm'],
+          })
+        ),
+      ]
+    }
+    const onHideConfirm = () => {
+      if (!stellarConfirmPaymentModalShown) {
+        return
+      }
+      stellarConfirmPaymentModalShown = false
```

Comment:
i actually meant checking the routeState. when this modal is up you can't do anything else? i'd lean towards removing this boolean. seems error prone